### ISSUE TITLE
Feature Importance

### ISSUE THREAD

**Issue by xnuohz** [pyg-team/pytorch-frame#282]
## Feature

Support feature importance in tabular data scenarios.
1. Understand which features are beneficial for prediction and help to develop new features
2. Feature selection, removing features that are not helpful in prediction

## Ideas

1. GBDTs naturally have APIs for calculating feature importance, it's easy to add.
2. NNs
    - Permutation. After shuffling a certain feature, observe the changes in metric. The greater the change, the more important the feature is. Simple.
    - [SHAP](https://github.com/shap/shap). Complex.

**Comment by yiweny** [comment]
Mutual Information Sort is already added [here](https://github.com/pyg-team/pytorch-frame/blob/master/torch_frame/transforms/mutual_information_sort.py). 
For feature sorting in NNs, I recommend you take a look at the ExcelFormer [example](https://pytorch-frame.readthedocs.io/en/latest/generated/torch_frame.nn.models.ExcelFormer.html#torch_frame.nn.models.ExcelFormer). 
If you are interested in adding any feature related functionalities, you can add it in the transform module.

**Comment by xnuohz** [comment]
Thanks. As you mentioned, `mutual information sorting` and `ExcelFormer` improve performance through transformation capabilities. However, I want to discuss how much different features contribute to the final prediction result. For example, user behavioral features are important in recommender systems, so their feature importance should be high. pytorch-frame is good to use. It allows me to quickly obtain benchmark results on real-world datasets to determine whether NNs or GBDTs are better. I'm unsure if the functionality to evaluate feature importance is worth integrating as a module into pytorch-frame.

**Comment by zechengz** [comment]
I think you can use Captum https://captum.ai/
cc @weihua916 we can also integrate this in PyT?

**Comment by xnuohz** [comment]
Yes, Captum implemented many interpretability methods, Feature Permutation and SHAP are part of them.

**Comment by February24-Lee** [comment]
Is there any update or roadmap related to it? 👀
